test/algorithms/searches/binary_search_test.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:DDSA/algorithms/searches/binary_search.dart';
+import 'package:test/test.dart';
+
+void main() {
```

Comment:
Awesome recycling of unit tests from the linear search from your past PR! I wonder if for search algorithms we can make this unit test suite generic since all the list based search algorithms will adhere to the same method signature/contract allowing us to not have to duplicate the test code and instead just use the single test suite and then in `ddsa_test` we just pass in a search function that adheres to a typedef function, i.e.
```
typedef SearchFunction = int Function<T>(List<T> list, T find);
```
But this is out of scope of this PR, so something we can do in a followup PR!